node/consensus/src/lib.rs

```diff
@@ -216,6 +222,9 @@ impl<N: Network> Consensus<N> {
         {
             metrics::increment_gauge(metrics::consensus::UNCONFIRMED_SOLUTIONS, 1f64);
             metrics::increment_gauge(metrics::consensus::UNCONFIRMED_TRANSMISSIONS, 1f64);
+            self.transmissions_queue_timestamps
+                .lock()
+                .insert(TransmissionID::Solution(solution.commitment()), snarkos_node_bft::helpers::now());
```

Comment:
it might be a good idea to obtain the timestamp before inserting into the collection; that would reduce the duration of the lock guard